docs/network_discovery.md

```diff
@@ -9,6 +9,18 @@ Home Assistant has built-in helpers to support mDNS/Zeroconf and SSDP. If your i
 
 ## Bluetooth
 
+### Best practices for library authors
+
+- When connecting to Bluetooth devices with `BleakClient`, always use the `BLEDevice` object instead of the `address` to avoid the client starting a scanner to find the `BLEDevice`. Call the `bluetooth.async_ble_device_from_address` API if you only have the `address`.
+
+- Call the `bluetooth.async_get_scanner` API to get a `BleakScanner` instance and pass it to your library. The returned scanner avoids the overhead of running multiple scanners, which is significant. Additionally, the wrapped scanner will continue functioning if the user changes the Bluetooth adapter settings.
```

Comment:
This is only necessary if the backend library uses a scanner, i.e., if there is no need to perform a scan, it is good enough to pass the bledevice for the backend library to use inside `(async_)update()`, right?

So step-wise the best practice is something like the following:
1. In `update()`, call `bluetooth.async_ble_device_from_address` to obtain a `BLEDevice` instance for the wanted addres.
2. Pass that instance to the upstream's `update()`, which should not re-use the `BleakClient` but create a create a new one and use that to establish the connection.
3. After querying the device, close the connection and free the used `BleakClient` instance.

What remains unclear to me is whether it's better to use the callback to get updates from homeassistant for the upcoming update cycle, or to use `async_ble_device_from_address` instead.

Reply:
I added answers below via email. It would be great if you did a followup to this PR to adjust based on your experience and perspective. I tried to encapsulate all the pitfalls we had developing for aiohomekit and the Ble sensors, but it would be good to get another perspective.